python_modules/dagster/dagster/core/asset_defs/time_window_partition_mapping.py

```diff
@@ -0,0 +1,96 @@
+from datetime import datetime, timedelta
+
+from dagster import check
+from dagster.core.definitions.partition import ScheduleType
+from dagster.core.definitions.time_window_partitions import TimeWindowPartitionsDefinition
+from dagster.core.errors import DagsterInvalidDefinitionError
+
+from .partition_key_range import PartitionKeyRange
+from .partition_mapping import PartitionMapping
+
+
+class TimeWindowPartitionMapping(PartitionMapping):
+    """
+    The default mapping between two TimeWindowPartitionsDefinitions.
+
+    A partition in the child partitions definition is mapped to all partitions in the parent whose
+    time windows overlap it.
+
+    This means that, if the parent and child partitions definitions share the same time grain, then
+    this mapping is essentially the identity partition mapping - plus conversion of datetime formats.
```

Comment:
Is grain a commonly accepted term?  had never heard it before.

I guess if we wanted to support offsets, we would do it in the `_map_partitions` function?

Reply:
Yeah, not really a commonly accepted term. I'm going to try changing it to "period".  Let me know if you think that's still confusing.